theories/common.elpi

```diff
@@ -140,7 +140,7 @@ quote.nat _ {{ Nat.of_num_uint lp:In }} {{ NC lp:OutM }} Out _ :-
   n-constant N OutM,
   quote.expr.constant { z-constant N } Out.
 quote.nat R In {{ NX lp:In }} Out VarMap :- !,
-  Zmodule = {{ GRing.Ring.zmodType lp:R }},
+  Zmodule = {{ GRing_Ring__to__GRing_Zmodule lp:R }},
```

Comment:
Although this might be an acceptable quick fix, it would be better to avoid explicitly mentioning such auto-generated names.

Reply:
Sure, let's see if we find a nicer solution before math-comp/math-comp#733 is merged

Reply:
Is it possible to declare an abbreviation (in `ring.v`) for this subtyping function without explicitly mentioning its name? @gares @CohenCyril

Reply:
I think that, for now, a `(*for algebra tac*) Notation xxx := GRing_Ring__to__GRing_Zmodule` in MC is the best option.

Reply:
`(id : ringType -> zmodType)`?

Reply:
Where? because {{ terms }} are not typed

Reply:
If we somehow look up such names from a database, it should be done *statically* (when or before compiling this Elpi file). Therefore, `(id : ringType -> zmodType)` is not a good solution. `Definition` may also introduce some (but very small, I guess) indirection in conversion.

Reply:
Is it possible to do `Register` in Elpi? Then it would be possible to refer to them using `lib:` in quotations from Elpi to Coq.

Reply:
I'm not sure I understand... anyway my point was that the best way to get a stable version of this is to ask Coq (regardless of the way) for the shortest coercion from `ringType` to `zmodType`.

Reply:
Yes, `Register` works in tandem with `lib:`

Reply:
Sorry, there is no API for `Register` in Elpi, but as usual I can add it for this use case.
Since I'm busy now, please open an issue and I'll do it after Decmber 12